ui/components/component-library/select-button/select-button.scss

```diff
@@ -1,13 +1,55 @@
 .mm-select-button {
-  box-sizing: border-box;
-  // size settings
-  &--type-danger {
-    // placeholder
+  --select-button-height: var(--size, 40px);
+
+  position: relative;
+  cursor: pointer;
+  text-align: start;
+  height: min-content;
+  min-height: var(--select-button-height);
+  overflow: hidden;
+  max-width: 100%;
+
+  &--size-sm {
+    --size: 32px;
+  }
+
+  &--size-md {
+    --size: 40px;
+  }
+
+  &--size-lg {
+    --size: 48px;
+  }
+
+
+  &--block {
+    width: 100%;
+  }
+
+  &:hover {
+    background-color: var(--color-background-hover);
+  }
+
+  &:active {
+    background-color: var(--color-background-pressed);
   }
 
   &:active,
   &:focus {
     border-color: var(--color-primary-default);
     box-shadow: inset 0 0 0 1px var(--color-primary-default);
   }
+
+  &--type-danger {
+    border-color: var(--color-error-default);
+    box-shadow: inset 0 0 0 1px var(--color-error-default);
+  }
+
+  &__content {
+    overflow: auto;
+  }
+}
+
+.mm-select-wrapper:has(.mm-select-button--block) {
+  width: 100%;
 }
```

Comment:
Here we're using the `mm-select-wrapper` className from `SelectWrapper` in another component's stylesheet. If we ever change that className in SelectWrapper, we'd need to remember to update it here too. Right now, we don't have a test to catch this. Is there a more stable way we can approach this?

Reply:
Ok as long as we've considered the trade offs.